src/pages/index.tsx

```diff
@@ -7,76 +7,55 @@ import { resultType } from "../utils/resultType";
 import React from "react";
 
 const Home: React.FC<{ results: resultType[] }> = ({ results }) => {
+
+  results.sort((a, b) => b.score - a.score);
+
   return (
-    <div className={styles.container}>
+    <main className={styles.container}>
       <Head>
         <title>MICP Leaderboard | TPC</title>
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />  
         <link rel="icon" href="/favicon.ico" />
       </Head>
       <Heading />
-      <table className={styles.table}>
-        <tbody>
-          <TableHeadings />
-          {results.map((e, i) => (
-            <TableRow key={i} {...e} />
-          ))}
-        </tbody>
-      </table>
-    </div>
+      <section className={styles.rankSection}>
+        {results.map((e, i) => (
+          <TableRow
+            key={i}
+            rank={i+1}
+            username={e.username}
+            name={e.name}
+            score={e.score}
+            currentRating={e.currentRating}
+          />
+        ))}
+      </section>
+    </main>
   );
 };
 
-export async function getStaticProps() {
-  const body = {
-    query:
-      "query {allMembers(_size: 1000) {data {username, initialRating, currentRating}}}",
-  };
-  const res = await fetch(process.env.DB_URL!, {
-    method: "POST",
-    headers: {
-      Authorization: "bearer " + process.env.DB_SECRET,
-    },
-    body: JSON.stringify(body),
-  });
-  const rawData = await res.json();
-  if (!rawData) {
+export async function getServerSideProps() {
+  const response = await fetch('https://micp-backend.onrender.com/api/rankList', {
+    method: 'GET',
+    mode: 'cors',
+    headers: {'Content-type' : 'application/json'}
+  })
+
+  if(!response.ok) {
     return {
-      notFound: true,
-    };
+      props: {}
+    }
   }
-  const arr: resultType[] = [];
-  const members = rawData.data.allMembers.data;
-  members.sort((a: any, b: any) => {
-    return (
-      b.currentRating - b.initialRating - (a.currentRating - a.initialRating)
-    );
-  });
-  let curRank = 1;
-  const map = new Map();
-  for (let i = 0; i < members.length; i++) {
-    const tempScore = members[i].currentRating - members[i].initialRating;
-    let tempRank = curRank;
-    if (map.has(tempScore)) {
-      tempRank = map.get(tempScore);
-    } else {
-      map.set(tempScore, tempRank);
+
+  const data = await response.json()
+  const rankList = data.users
+
+  if(data) {
+    return {
+      props : {
+        results : rankList
+      }
```

Comment:
Why are we not caching these results? The response will be the same unless there is some contest.

Reply:
I am using cache control for caching...
since the site is deployed on vercel I thought of using VEN..
the data will be refreshed once every 3days